modular_skyrat/modules/modular_items/lewd_items/code/lewd_items/venom_milker.dm

```diff
@@ -0,0 +1,115 @@
+/obj/item/reagent_containers/venom_milker
+	name = "\improper venom siphon"
+	desc = "A commercial grade venom siphon, made for use on larger - typically human sized - animals. \
+	Has a built-in reagent neutralizer that inhibits the effects of most extracted toxins for safe handling, but it cannot be guaranteed to work."
+
+	icon_state = "venom_milker"
+	icon = 'modular_skyrat/modules/modular_items/lewd_items/icons/obj/lewd_items/lewd_items.dmi'
+
+	volume = 20
+
+	reagent_flags = OPENCONTAINER
+	w_class = WEIGHT_CLASS_SMALL
+
+/obj/item/reagent_containers/venom_milker/Initialize(mapload)
+	. = ..()
+
+	var/milkable_string = /datum/preference/choiced/venomous_bite_venom::milkable_string
+	if (length(milkable_string))
+		desc += span_notice("\nThe following reagents cannot be filtered by the neutralizer: [milkable_string]")
+
+/obj/item/reagent_containers/venom_milker/attack(mob/living/target_mob, mob/living/user, params)
+	. = ..()
+
+	if (!can_milk(target_mob, user))
+		return FALSE
+
+	playsound(user, 'sound/effects/compressed_air/tank_insert_clunky.ogg', 50)
+	user.balloon_alert_to_viewers("siphoning...")
+
+	var/text = span_purple("[user] starts hooking up [src] to [target_mob]'s fangs...")
+	var/self_text = span_purple("You start hooking up [src] to [target_mob]'s fangs...")
+	var/victim_text = span_purple("[user] starts hooking up [src] to your fangs...")
+
+	user.visible_message(text, self_text, ignored_mobs = target_mob)
+	to_chat(target_mob, victim_text)
+
+	if (!do_after(user, 3 SECONDS, target_mob, IGNORE_HELD_ITEM, extra_checks = CALLBACK(src, PROC_REF(can_milk), target_mob, user)))
+		return FALSE
+	playsound(user, 'sound/effects/compressed_air/tank_remove_thunk.ogg', 50)
+
+	siphon(target_mob, user)
+
+	return TRUE
+
+/**
+ *
+ * Checks if we can milk the target. Returns TRUE/FALSE.
+ *
+ * Args:
+ * * mob/living/target: The target. Non-nullable.
+ * * mob/living/user: The user. Nullable.
+ * * silent = FALSE: If TRUE, will not give user any feedback.
+ */
+/obj/item/reagent_containers/venom_milker/proc/can_milk(mob/living/target, mob/living/user, silent = FALSE)
+	var/datum/action/cooldown/mob_cooldown/venomous_bite/bite = locate(/datum/action/cooldown/mob_cooldown/venomous_bite) in target.actions
```

Comment:
Remove the type in locate, it's unnecesary, byond just uses the type in var/ when it's set